src/sdk.ts

```diff
@@ -440,7 +440,49 @@ export class DefaultRelayingServices implements RelayingServices {
         return estimate.toString();
     }
 
-    getAccountAddress(): string {
+    async estimateMaxPossibleRelayGasWithLinearFit(
+        destinationContract: string,
+        smartWalletAddress: string,
+        tokenFees: string,
+        abiEncodedTx: string,
+        relayWorker: string
+    ): Promise<number> {
+        console.debug('estimateMaxPossibleRelayGasWithLinearFit Params', {
+            destinationContract,
+            smartWalletAddress,
+            tokenFees,
+            abiEncodedTx,
+            relayWorker
+        });
+
+        const gasPrice = toBN(
+            await this.relayProvider.relayClient._calculateGasPrice()
+        );
+        const tokenAmount = await this.web3Instance.utils.toWei(tokenFees);
+        const trxDetails = {
+            from: this._getAccountAddress(),
+            to: destinationContract,
+            value: '0',
+            relayHub: this.contracts.addresses.relayHub,
+            callVerifier: this.contracts.addresses.smartWalletRelayVerifier,
+            callForwarder: smartWalletAddress,
+            data: abiEncodedTx,
+            tokenContract: this.contracts.addresses.testToken,
+            tokenAmount: tokenAmount,
+            onlyPreferredRelays: true
+        };
+
+        const maxPossibleGasValue =
+            await this.relayProvider.relayClient.estimateMaxPossibleRelayGasWithLinearFit(
+                trxDetails,
+                relayWorker
+            );
+        const maxPossibleGas = toBN(maxPossibleGasValue);
+        const estimate = maxPossibleGas.mul(gasPrice);
+        return Number(estimate.toString());
+    }
+
+    private _getAccountAddress(): string {
```

Comment:
is there any reason why this function was renamed?

Reply:
getAccountAddress was rename to avoiding confusion because it is a private method.

Reply:
i don't think it was necessary, but it's not a blocker

(see [here](https://stackoverflow.com/questions/47747809/typescript-underscore-convention-for-members))